### PR TITLE
Reworked logging trait dependencies

### PR DESCRIPTION
**Pull request by orpiske**
As part of GH issue apache/camel-k#2541, moved the dependency to the runtime to ensure it's always present, even with custom base images or other customizations. This also allow us to manage the dependencies on a single place instead of spreading them over the codebase.

Depends on apache/camel-k-runtime#731

**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Comment by orpiske** [comment]
Added bonus of this is that the container does not required a rebuild if the `logging.json` trait is turned on or off.

File: pkg/trait/logging.go [hunk 1]
```diff
@@ -73,7 +71,6 @@ func (l loggingTrait) Apply(environment *Environment) error {
 			if environment.Integration.Status.Dependencies == nil {
 				environment.Integration.Status.Dependencies = make([]string, 0)
 			}
-			util.StringSliceUniqueAdd(&environment.Integration.Status.Dependencies, depQuarkusLoggingJson)
```
Comment (astefanutti) [inline pkg/trait/logging.go:76]: It seems like the whole code block can be removed?

Comment (orpiske) [reply pkg/trait/logging.go:76]: Thanks! Yes it can be removed.

File: pkg/trait/logging.go [hunk 2]
```diff
@@ -68,17 +66,6 @@ func (l loggingTrait) Configure(environment *Environment) (bool, error) {
 }
 
 func (l loggingTrait) Apply(environment *Environment) error {
-	if environment.IntegrationInPhase(v1.IntegrationPhaseInitialization) {
```
Comment (astefanutti) [inline pkg/trait/logging.go:71]: I think running the trait during the `IntegrationPhaseInitialization` phase is not needed anymore :)

Comment (orpiske) [reply pkg/trait/logging.go:71]: ... and it's gone! :D 

Thanks!

**Review by astefanutti — APPROVED** [review]

**Comment by orpiske** [comment]
So, I think we need to wait before merging this because it depends on Camel K Runtime 1.9. We are currently defaulting to 1.8.0 on the Makefile.

**Pull request merged**